### PR TITLE
feat: new branch for absolute path closes #5484

### LINKED ISSUE #5484: Replace relative imports to `@`
Our current `tsconfig` allows all files compiled via TypeScript to use `@` to indicate the `root` folder.

This allows us to replace long relative imports such as `import something from '../../../../big-path/something';` to simply `import something from '@/big-path/something';`

This PR is about updating all these relative paths with `@` by doing the work manually or having an ESLint rule from `.ts/.tsx` files to suggest that and fix that automatically.

### PR DESCRIPTION
**Pull request by umairraza96**
### Check List

- [x] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [x] I have run `npx turbo lint` to ensure the code follows the style guide. And run `npx turbo lint:fix` to fix the style errors if necessary.
- [x] I have run `npx turbo format` to ensure the code follows the style guide.
- [x] I have run `npx turbo test` to check if all tests are passing, and/or `npx turbo test:snapshot` to update snapshots if I created and/or updated React Components.
- [x] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

File: .eslintrc
Comment (ovflowd) [inline .eslintrc]: I see that some unrelated lines tot changed? Is that maybe your IDE? Or something applying some custom format?

File: components/Article/Codebox/__snapshots__/index.stories.tsx.snap
Comment (ovflowd) [inline components/Article/Codebox/__snapshots__/index.stories.tsx.snap:1]: Any reason why this snapshot got updated? (Actually any reason at all why you ran the storybook snapshot command? 🤔)

File: package.json
Comment (ovflowd) [inline package.json]: Please read the DEPENDENCY_PINNING.md, this dependency should use an exact version, instead of “^”

**Review by ovflowd — COMMENTED** [review]
Overall looks good, but before I give an approval I left a few comments regarding a few concerns.

Again thank you for your contribution! ✨

File: components/Article/Codebox/__snapshots__/index.stories.tsx.snap
Comment (umairraza96) [reply components/Article/Codebox/__snapshots__/index.stories.tsx.snap:1]: `npx turbo test:storybook` was failing for me and it was some article snapshot that was erroring since I didn't updated that component I figured running snapshot wouldn't hurt

Comment (umairraza96) [reply components/Article/Codebox/__snapshots__/index.stories.tsx.snap:1]: since the snapshot is updated it was evident that there was something wrong otherwise snapshots shouldn't update right ?

File: .eslintrc
Comment (umairraza96) [reply .eslintrc]: I'll check on that I am using `neovim` and may be my `json-lsp` is not aligned with that, what are you guys using to format this ?

Comment (umairraza96) [reply .eslintrc]: Figured this one out, turns out you guys are using `prettier` for formatting you `json`

Comment (ovflowd) [reply .eslintrc]: Sure, but I’ve edited this file countless times. Why it only got reformatted by you? 🙃

**Comment by umairraza96** [comment]
@ovflowd I believe that all the formatting should be included in CI/CD pipepline hook or at least in our precommit, in this way different dev environments won't bother in formatting code

File: components/Article/Codebox/__snapshots__/index.stories.tsx.snap
Comment (ovflowd) [reply components/Article/Codebox/__snapshots__/index.stories.tsx.snap:1]: No, just think with me right here, if the snapshot was failing, how is the main branch and all other PRs green? 😉

You might have encountered an unrelated flakiness. But please if you might revert this snapshot? We don’t want unrelated changes here.

If after reverting CI starts failing then we have a flaky test.

**Comment by ovflowd** [comment]
@umairraza96 we already have linting and formatting checks on our CI pipelines and in our precommit hook. I appreciate the concern, but the issue is definitely on your side.

File: .eslintrc
Comment (ovflowd) [reply .eslintrc]: To be more precise here, the “.rc” files (or at least a few of them) are indeed JSON files but they don’t have a JSON file extension.

So it is erroneous for prettier to simply format it out of the blue (except if you have prettier enabled for that file)

Our pre-commit hook definitely will not format those files, and that could theoretically be “solved” if we applied JSON extensions to these files (cc @bmuenzenmeyer, as I sadly think we might want to do that sooner than GHC)

Since the file is not being linted/formatted by precommit hooks and has no apparent file extension, its formatting will clearly depend on how your IDE decides to format that file.

That can be a combination of extensions and sort of IDE you have.

Comment (ovflowd) [reply .eslintrc]: TL;DR, this file shouldn’t have been formatted, and yes, we are aware that this flakiness of formatting is due to the file not being actually handled by precommit rules and there are other things at play, there’s an open issue for that.

But it is honestly too much draining from my side to explain every little detail for a newcomer on their first PR. Hence **why** I’m simply asking you to follow my recommendations, as I would expect you to expect that an maintainer of an open source project to know (to a certain extent) what they’re talking about 😅

File: components/Article/Codebox/__snapshots__/index.stories.tsx.snap
Comment (ovflowd) [reply components/Article/Codebox/__snapshots__/index.stories.tsx.snap:1]: And if that’s the case, well, I have an idea what the flakiness is about, and was probably introduced on a recent PR by @araujogui where the Codebox component syntax highlighting was already asynchronous but now might be taking enough miliseconds for it to at sometimes being not “re-rendered” at the time of the snapshot.

@araujogui you might want to open a PR to ensure that we wait the component to have its syntax highlighting executed before taking the snapshot. Not sure how to do that with Storybook, so you might want to do some 👀👀

File: .eslintrc
Comment (umairraza96) [reply .eslintrc]: its just I had a different plugin/extension which was taken precedence over prettier

Comment (ovflowd) [reply .eslintrc]: Sorry for the big texts @umairraza96 just trying to give you some context here 🙇🙏

Comment (umairraza96) [reply .eslintrc]: I don't mind at all, I deeply appreciate the level of details you shared with me :)

**Comment by ovflowd** [comment]
@umairraza96 https://github.com/nodejs/nodejs.org/pull/5514 this PR might solve your issues 🙇

File: .eslintrc
Comment (ovflowd) [reply .eslintrc]: > its just I had a different plugin/extension which was taken precedence over prettier

Yeah, that makes sense, and `lint-staged` would be fixing that, **if** the `.rc` files had proper file extensions.

**Comment by umairraza96** [comment]
@ovflowd let me know if anything is required from my side.

**Comment by ovflowd** [comment]
Absolutely! Sorry, just relaxing this weekend! Ill look your PR tomorrow!

**Review by ovflowd — APPROVED** [review]
LGTM! Thanks for the contribution and for the patience!

**Comment by ovflowd** [comment]
And landed :shipit: 🚢 🎉

**Pull request merged**
